### DISCUSSION TITLE
Building simdjson DLL in Visual Studio (2022) not working ?

### DISCUSSION THREAD

**Discussion by crashXpert** [simdjson/simdjson#2194] [General]
Hello,

I try to make a simple Console App in msdev executing simdjson example of section **"Output to strings"** in https://github.com/simdjson/simdjson/blob/master/doc/basics.md#using-the-parsed-json (see _TEST_cpp.txt_ joined)

To generate the dll project, I have follow **"Usage (CMake on Windows using Visual Studio)"** section in https://archive.org/details/github.com-lemire-simdjson_-_2019-03-08_11-15-39 

Building project **ALL_BUILD** generate the simdjson.dll, but I can't see nowhere in the generated solution the singleheader file simdjson.h.
Moreover, there is no project **RUN_TESTS**  in the solution. May be due to some error in cmake commande (see _CMAKE_Trace.txt_ joined)

When I Build my Console App , simdjson.h file is not found, even if I add it to the project ?

What is the good way to achieve this simple use case ?

Thanks

**Comment by crashXpert** [comment]
[CMAKE_Trace.txt](https://github.com/user-attachments/files/15737958/CMAKE_Trace.txt)
[TEST_cpp.txt](https://github.com/user-attachments/files/15738166/TEST_cpp.txt)

**Comment by lemire** [comment]
Many Visual Studio users will want to use vcpkg, and we make a complete example available:

https://github.com/simdjson/simdjson-vcpkg/tree/master

**Reply by crashXpert** [reply]
OK, thanks I will try.

**Comment by lemire** [comment]
>  I can't see nowhere in the generated solution the singleheader file simdjson.h.

You can grab the single-header files from our releases (simdjson.h and simdjson.cpp):

https://github.com/simdjson/simdjson/releases/tag/v3.9.3

You can drop these files in your project and be good to go. You do not need to use CMake at all unless you want to. If you want to use CMake with Visual Studio, then [I recommend using vcpkg](https://github.com/simdjson/simdjson-vcpkg/tree/master).

**Reply by crashXpert** [reply]
Thanks, but to be honest I have try to build my own DLL from single-header files (simdjson.h and simdjson.cpp). The DLL is welle generated and I can used it, except that I have LINK Error as soon as I use get_double()  (impossible to resolve simdjson::internal::from_chars). I have no problem using get_int64() for example. I have to make my own service using to_json_string and std::from_chars to get a double. Obviously, I don't want to make any override and be able to use simdjson.h and simdjson.cpp as is. It's why I  try another approach to generate dll.

**Reply by lemire** [reply]
@crashXpert Instead of building a DLL, just build simdjson.cpp along with the rest of your code.

**Reply by lemire** [reply]
Or just use [vcpkg](https://github.com/simdjson/simdjson-vcpkg/tree/master). It was designed to make it easy to add dependencies.

**Reply by crashXpert** [reply]
Thanks, but I have to use a dll for modularity architecture purpose (simdjson used sharded between several component).
I will have a look to vcpkg. I don't know this package manager.
However, I'm very interesting to know if someone else faces this issue, and how he has resolved it.

**Comment by lemire** [comment]
**Please do not refer to documentation from five years ago, prior to our 1.0 release.**

**Reply by crashXpert** [reply]
OK, no problem, I'm understand.
But in this case, I haven't found any more recent post on this subject.
Sorry for that.
thanks
